src/Sanitize.php

```diff
@@ -652,20 +647,30 @@ private function get_cache(string $image_url = ''): DataCache
     private function get_http_client(): Client
     {
         if ($this->http_client === null) {
-            return new FileClient(
+            $this->http_client = new FileClient(
```

Comment:
It's possible to change the parameters with `pass_file_data()` multiple times. Caching the FileClient would not reflect this changes in the FileClient and is a BC break. Alternatively we could unset $this->http_client inside `pass_file_data()`.

Reply:
Good catch, did the latter.

Reply:
We have the same problem with class `SimplePie` and the methods `set_timeout()`, `set_curl_options()`, `force_fsockopen()` and `set_useragent()`. I have solved this issue in #777 by unset the http_client or trigger an error, if a PSR-18 client was provided.

```php
    public function force_fsockopen(bool $enable = false)
    {
        $this->force_fsockopen = $enable;

        // Reset a possible existing FileClient,
        // so a new client with the changed value will be created
        if (is_object($this->http_client) && $this->http_client instanceof FileClient) {
            $this->http_client = null;
        } elseif (is_object($this->http_client)) {
            // Trigger notice if a PSR-18 client was set
            trigger_error(sprintf(
                'Using "%s()" has no effect, because you already provided a HTTP client with "%s::set_http_client()". Configure fsockopen in your HTTP client instead.',
                __METHOD__,
                get_class($this)
            ), \E_USER_NOTICE);
        }
    }
```

Reply:
Thanks, I made it a hard error since here user would have to explicitly switch to `set_http_client`.

Reply:
Fair enough. 😊 
But we has also to unset a possible created $this->http_client on calling` set_timeout()`, `set_curl_options()`, etc.